### DISCUSSION TITLE
Issue with $at_timezone function in timetz

### DISCUSSION THREAD

**Discussion by gureshme** [trinodb/trino#16679] [General]
I have come across this error in trino which is throwing error when I run query with select at_timezone(current_time,"+-4") but same thing works when I run at_timezone(current_timestamp,"+-4") or when I pass $ with argument it also works like -select at_timezone($current_time,"+-4"). I see value diff between function at_timezone in files files timetz/AtTimezone and timestamptz/AtTimezone. In timetz at_timezone function has value @ScalarFunction(value = "$at_timezone", hidden = true) whereas in timestamptz/AtTimezone there is no $ sign and hidden true. I wanted to understand the significance of $ in timetz/AtTimezone and hidden=true.

**Comment by martint** [comment]
The $at_timezone function is an internal engine function used to implement the `AT TIME ZONE ...` syntax. It's not meant to be used directly, and may be removed or changed in the future without notice.
